src/dist/replication/test/replica_test/unit_test/replica_test.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include <gtest/gtest.h>
+
+#include <dsn/utility/fail_point.h>
+#include "replica_test_base.h"
+
+namespace dsn {
+namespace replication {
+
+class replica_test : public replica_test_base
+{
+public:
+    int total_count = 0;
+    dsn::app_info _app_info;
+    dsn::gpid pid = gpid(2, 1);
+
+public:
+    void SetUp() override
+    {
+        stub->install_perf_counters();
+        mock_app_info();
+        stub->generate_replica(_app_info, pid, partition_status::PS_PRIMARY, 1);
+    }
+
+    void calc_write_size_exceed_threshold_count()
+    {
+        total_count += stub->_counter_recent_write_size_exceed_threshold_count->get_value();
+    }
+
+    void mock_app_info()
+    {
+        _app_info.app_id = 2;
+        _app_info.app_name = "replica_test";
+        _app_info.app_type = "replica";
+        _app_info.is_stateful = true;
+        _app_info.max_replica_count = 3;
+        _app_info.partition_count = 8;
+    }
+};
+
+TEST_F(replica_test, write_size_limited)
+{
+    int count = 100;
+    task_code default_code;
+    struct dsn::message_header header;
+    header.body_length = 10000000;
+
+    auto write_request = dsn::message_ex::create_request(default_code);
+    write_request->header = &header;
+    while (count-- > 0) {
+        stub->on_client_write(pid, write_request);
+        calc_write_size_exceed_threshold_count();
```

Comment:
I think you can move ```calc_write_size_exceed_threshold_count()``` out of while

Reply:
No, I have test it. `_counter_recent_write_size_exceed_threshold_count` value is not "accumulative value"(it equal 0 or 1), you need accumulate it every time.